### PR TITLE
feat(stageleft): re-compile staged sources for the macro at the top level

### PR DESCRIPTION
**Pull request by shadaj**
feat(stageleft): re-compile staged sources for the macro at the top level

Summary:

Test Plan:

### CODE REVIEW

File: stageleft/README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ name = "foo_macro"
 
 [lib]
 proc-macro = true
-path = "src/lib.rs"
```
Comment (MingweiSamuel) [inline stageleft/README.md:81]: this is a bit of a hack?

File: hydroflow_plus_test/src/lib.rs [hunk 2]
```diff
@@ -6,9 +6,24 @@ use hydroflow_plus::tokio_stream::wrappers::UnboundedReceiverStream;
 use hydroflow_plus::*;
 use stageleft::{q, Quoted, RuntimeData};
 
+#[cfg(feature = "macro")]
+pub(crate) mod cluster;
+#[cfg(not(feature = "macro"))]
 pub mod cluster;
```
Comment (MingweiSamuel) [inline hydroflow_plus_test/src/lib.rs:12]: These seem backwards? features need to be additive?

Comment (shadaj) [reply hydroflow_plus_test/src/lib.rs:12]: This is related to above hack, we want to not have `pub mod` when we are compiling the top level `lib` for a macro crate.

**Review by MingweiSamuel — APPROVED** [review]
Would it be possible to use a cargo cfg instead of a crate feature `macro`?

Like with proc-macro2: `RUSTFLAGS='--cfg procmacro2_semver_exempt' cargo build`

**Pull request merged**